backend/pkg/console/topic_overview.go

```diff
@@ -63,7 +65,19 @@ func (s *Service) GetTopicsOverview(ctx context.Context) ([]*TopicSummary, error
 		return nil, err
 	}
 
-	topicNames := metadata.Topics.Names()
+	// 2. Extract all topicNames from metadata
+	topicNames := make([]string, 0, len(metadata.Topics))
+	for _, topic := range metadata.Topics {
+		topicName := topic.Topic
+		if topic.Err != nil {
+			s.logger.Error("failed to get topic metadata while listing topics",
+				zap.String("topic_name", topicName),
+				zap.Error(topic.Err))
+			return nil, topic.Err
+		}
+
+		topicNames = append(topicNames, topicName)
+	}
```

Comment:
Currently in master we return an error if any topic metadata requests failed (see [code](https://github.com/redpanda-data/console/blob/master/backend/pkg/console/topic_overview.go#L171-L177)). This brings the same logic back in. Alternatively I suppose we can still return the topics that succeeded, and not fail the request. If so, we need to adjust tests appropriately.